### ISSUE TITLE
Unification of handling weights passed as arguments to functions 

### ISSUE THREAD

**Issue by msluszniak** [elixir-nx/scholar#249]
As mentioned [here](https://github.com/elixir-nx/scholar/pull/247#issuecomment-2038130186) we want to unify the way we pass the weights in Scholar to the algorithms. Currently, there are a few ways that we do it. For example, we pass them as tensors or as a list via `opts`.

**Comment by JoaquinIglesiasTurina** [comment]
I would like to take a stab at this one, particularly regarding linear methods.
I think this would also be a great opportunity to DRY up some of the duplication of private functions that's spreading over there.
So far, the better approach I've found is to move all these functions to a helper module, make them public,
but make sure to set `@moduledoc false` and `@doc false`. My proof of concept is [here](https://github.com/JoaquinIglesiasTurina/scholar/blob/main/lib/scholar/linear/linear_helpers.ex).
Is this the best approach to handle this? Or should I take an alternative approach?

**Comment by msluszniak** [comment]
Yes, this looks great! You can send a PR with your improvements :)
